src/ngraph/op/reshape.cpp

```diff
@@ -298,3 +360,135 @@ void op::v1::Reshape::generate_adjoints(autodiff::Adjoints& /* adjoints */,
 {
     throw ngraph_error("generate_adjoints not implemented for Reshape");
 }
+
+bool op::v1::Reshape::evaluate(const HostTensorVector& outputs, const HostTensorVector& inputs)
+{
+    // infer and set output shape if the output shape contain -1
+    // and zero value dimension
+    size_t output_rank = inputs[1]->get_shape()[0];
+    Shape out_shape_val;
+
+#define COMPUTE_OUPUT_SHAPE(T)                                                                     \
```

Comment:
Use a template and base it on `element_type_traits` to ensure that the the C++ type is the correct one for the element type.